lib/matplotlib/contour.py

```diff
@@ -816,6 +816,11 @@ def __init__(self, ax, *args,
         self._extend_min = self.extend in ['min', 'both']
         self._extend_max = self.extend in ['max', 'both']
         if self.colors is not None:
+            if isinstance(self.colors, str):
```

Comment:
```suggestion
            mcolors.is_color_like(self.colors):
```
I'm wondering if this would allow us to drop the documented restriction that a single colour has to be specified as a string.  I'm not sure why that restriction was there: with v3.9.1 passing an RGB or RBGA tuple seems to work fine (as long as I don't pass _extend_).  It does currently break when passing a `(color, alpha)` pair, but I think that form is much newer than this restriction.

Reply:
This looks right. I'm also 95% confident there aren't any weird edge cases - I think they would require an expression that can be used both as single color and list of colors.